clients/python-wrapper/lakefs/object_io.py

```diff
@@ -221,29 +223,65 @@ def _extract_etag_from_response(headers) -> str:
         etag = headers.get("ETag", "").strip(' "')
         return etag
 
-    def _upload(self, content, pre_sign, content_type: Optional[str] = None,
-                metadata: Optional[dict[str, str]] = None) -> ObjectStats:
-        if not pre_sign:
-            raise UnsupportedOperationException("Upload currently supported only in pre-sign mode")
-
-        headers = {}
-        if content_type is not None:
-            headers["Content-Type"] = content_type
+    def _upload_raw(self, content, content_type, metadata):
+        """
+        Preform upload api using raw api call to bypass validation of content parameter
+        """
+        headers = {
+            "Accept": self._client.sdk_client.objects_api.api_client.select_header_accept(['application/json']),
+            "Content-Type": content_type if content_type is not None else "application/octet-stream"
+        }
+
+        # Create user metadata headers
+        for k, v in metadata.items():
+            headers[_LAKEFS_METADATA_PREFIX + k] = v
+
+        _response_types_map = {
```

Comment:
It doesn't since this map is specific per API and is hardcoded for each method

Reply:
Again, this is problematic: it means that version upgrades become more expensive.  If we upgrade the SDK, we need to upgrade the wrapper.

In any case, AFAICS `call_api` only uses its `response_types_map` argument when performing an async call,  Consider just adding a comment to add it if/when we go async.

Reply:
call_api uses the response_type_maps in both sync and async calls.
I'm not sure this is a big issue - if we change the return object from the upload object api call it will anyway be a  breaking change that we will need to deal with whether we use a raw api call or not.
The worst that can happen is that we will map errors codes that were removed or not map new error codes that were added (not a breaking change but will require a version update).
_The alternative_ to the hack is keeping our own http client, invoking a raw command parsing the status and the returned object all by ourselves - which is prone to the exact same issues

Reply:
But this **IS** used in sync calls...